### PR TITLE
feat: add reporter layer to support JSON output

### PR DESCRIPTION
**Pull request by vostrik**
Added  JSON output with `--json` argument.

Also added reporter layer to extend in future base reporters. It can be useful in integrations to such tools like Teamcity, etc.

#94

### CODE REVIEW

**Comment by ai** [comment]
I like your architecture. But let’s reduce the number of classes. `JsonReporter` and `HumanReporter` will be enough.

File: runner.js [hunk 1]
```diff
@@ -442,10 +450,10 @@ async function main () {
       }
       fix += chalk.bold(configPath)
     }
-    output += '\n  ' + chalk.yellow(fix)
+    hint = chalk.yellow(fix)
   }
 
-  process.stdout.write('\n' + output + '\n')
+  reporter.logResults({ results, hint })
```
Comment (ai) [inline runner.js]: You do not need objects here. Just use arguments `reporter.logResults(results, hint)`

File: runner.js [hunk 2]
```diff
@@ -486,6 +494,8 @@ main().catch(e => {
     msg = e.stack
   }
 
-  process.stderr.write(`${ chalk.bgRed(' ERROR ') } ${ chalk.red(msg) }\n`)
+  reporter.error({
+    message: `${ chalk.bgRed(' ERROR ') } ${ chalk.red(msg) }\n`
```
Comment (ai) [inline runner.js]: Should we move ` ERROR ` to a reporter?

Comment (vostrik) [reply runner.js]: Yeah, I think so. Also in `warn` method?

File: reporters/std-json-reporter.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  error ({ message }) {
+    super.error({
+      message: this.format({ message: { error: message } })
+    })
+  }
+}
+
+module.exports = {
```
Comment (ai) [inline reporters/std-json-reporter.js]: We can directly export class here

File: runner.js (hunk 2 above)
Comment (ai) [reply runner.js]: Yeap

File: reporters/std-json-reporter.js (hunk 3 above)
Comment (vostrik) [reply reporters/std-json-reporter.js]: Ok. 
Do you think that named export to save naming consistence in this case is bad practice?

Comment (ai) [reply reporters/std-json-reporter.js]: I am exporting function directly without object wrap https://github.com/ai/size-limit/blob/master/index.js#L255

So for consistency, we should do the same with new files.

File: runner.js (hunk 1 above)
Comment (vostrik) [reply runner.js]: Done in all methods

File: runner.js (hunk 2 above)
Comment (vostrik) [reply runner.js]: Done for `warn()` and `error()` in `HumanReporter`

File: reporters/std-json-reporter.js (hunk 3 above)
Comment (vostrik) [reply reporters/std-json-reporter.js]: Done for all new modules

**Comment by ai** [comment]
I simplified code even more 27891e6

**Pull request merged**